### PR TITLE
Add limited H100 testing for CUB 

### PR DESCRIPTION
**Pull request by jrhemstad**
Adds limited H100 per-PR CI testing for CUB.

### CODE REVIEW

**Review by gevtushenko — APPROVED** [review]

**Review by bernhardmgruber — APPROVED** [review]

File: ci/matrix.yaml [hunk 1]
```diff
@@ -8,6 +8,8 @@ workflows:
   #   - {jobs: ['test'], project: 'thrust', std: 17, ctk: 'curr', cxx: ['gcc12', 'llvm16']}
   #
   override:
+      - {jobs: ['test_lid0'], project: ['cub'],  gpu: 'h100',  sm: 'gpu', std: 'max', cxx: 'gcc12' }
```
Comment (miscco) [inline ci/matrix.yaml:11]: This overrides the current setting, so I assume this should not be merged

**Review by miscco — CHANGES_REQUESTED** [review]

File: ci/matrix.yaml (hunk 1 above)
Comment (jrhemstad) [reply ci/matrix.yaml:11]: Our CI status check already will block merging a PR when `override` is used :)

**Pull request merged**